test/AudioParam-test.js

```diff
@@ -400,31 +350,47 @@ describe('AudioParam', function() {
 
     it('should calculate the values for each sample if a-rate', function() {
       var audioParam = new AudioParam(dummyContext, 1, 'a')
-        , t1 = 12800 / 10 * Ts
-        , t2 = t1 + 12800 / 5 * Ts
-        , t3 = t2 + 12800 / 5 * Ts
-        , t4 = t3 + 12800 / 5 * Ts
-        , t5 = t4 + (12800 / 10 + 12800 / 5) * Ts
+        , t1 = 12800 / 4 * Ts
+        , t2 = t1 + 12800 / 4 * Ts
+        , t3 = t2 + 12800 / 4 * Ts
+        , t4 = t3 + 12800 / 4 * Ts
+        , t5 = (12800 / 4 + 12800 / 10) * Ts
         , block
 
       // Ramp t=0
       audioParam.setValueCurveAtTime([1, 2, 3, 4, 5], 0, 12800 * Ts)
       assert.equal(audioParam.value, 1)
       untilTime(audioParam, t1, function(block, Tb) {
         assertValidBlock(block)
-        assertAllValuesEqual(block.getChannelData(0), 1)
+        assertAllValuesFunc(block.getChannelData(0), Tb, function(t) {
+          var theoreticIndex = (t / (12800 * Ts)) * 4
+
+          return (1 - (theoreticIndex - 0)) * 1 + (1 - (1 - theoreticIndex)) * 2
+        })
       })
       untilTime(audioParam, t2, function(block, Tb) {
         assertValidBlock(block)
-        assertAllValuesEqual(block.getChannelData(0), 2)
+        assertAllValuesFunc(block.getChannelData(0), Tb, function(t) {
+          var theoreticIndex = (t / (12800 * Ts)) * 4
+
+          return (1 - (theoreticIndex - 1)) * 2 + (1 - (2 - theoreticIndex)) * 3
+        })
       })
       untilTime(audioParam, t3, function(block, Tb) {
         assertValidBlock(block)
-        assertAllValuesEqual(block.getChannelData(0), 3)
+        assertAllValuesFunc(block.getChannelData(0), Tb, function(t) {
+          var theoreticIndex = (t / (12800 * Ts)) * 4
+
+          return (1 - (theoreticIndex - 2)) * 3 + (1 - (3 - theoreticIndex)) * 4
+        })
       })
       untilTime(audioParam, t4, function(block, Tb) {
         assertValidBlock(block)
-        assertAllValuesEqual(block.getChannelData(0), 4)
+        assertAllValuesFunc(block.getChannelData(0), Tb, function(t) {
+          var theoreticIndex = (t / (12800 * Ts)) * 4
```

Comment:
This seems a bit too magical. Do you mind elaborating?

Reply:
Previously the `setValueCurveAtTime` automation was implement by applying the exact values from the curve in steps. But the spec says the values should be interpolated and for that I had to update the tests to expect the value be interpolated.

Does that make sense?

Reply:
Yeah, totally. Prob just worth mentioning in comment that interpolation is going on.

Reply:
I realized that in this specific case the computation is the same as for the `linearRampToValueAtTime` test. I updated the test.